### PR TITLE
 Add reference guide and source code sections in Japanese site.

### PR DESCRIPTION
**Pull request by itohiro73**
I also made some minor fixes in site files in second commit.

### CODE REVIEW

**Comment by nikhilnanivadekar** [comment]
@itohiro73 can you please cherry pick my commit from #493 and close that PR. I'll merge this PR.

**Comment by itohiro73** [comment]
@nikhilnanivadekar Sure, done.

File: site/ja/index.html [hunk 1]
```diff
@@ -523,13 +529,35 @@ <h2 class="section-heading">Eclipse Collectionsの歴史</h2>
...
 </div>
-<!-- Introduction  -->
+<!-- Reference Guide  -->
+
+<a class="anchor" name="sourceCode"></a>
+<div class="content-section-b">
+    <div class="container-fluid">
+        <h2 class="section-heading">ソースコード</h2>
+        <p class="lead">
+            ソースコードはGitHubにホストされています: <a href="https://github.com/eclipse/eclipse-collections">Eclipse Collections</a></p>.
```
Comment (nikhilnanivadekar) [inline site/ja/index.html]: the last `<\p>` tag is incorrect. That was one of the fixes I did for other files as a part of the commit you CP-ed.

File: site/cn/index.html [hunk 2]
```diff
@@ -549,7 +549,7 @@ <h2 class="section-heading">参考指南</h2>
         <div class="container-fluid">
             <h2 class="section-heading">源代码</h2>
             <p class="lead">
-                在GitHub上可以找到Eclipse Collections<a href="https://github.com/eclipse/eclipse-collections">的源代码</a>.
+                在GitHub上可以找到<a href="https://github.com/eclipse/eclipse-collections">Eclipse Collections</a>的源代码.
```
Comment (nikhilnanivadekar) [inline site/cn/index.html]: From Google Translate 的源代码 means Source Code. So I had switched the text to be consistent with others. But may be I got it wrong.

**Comment by itohiro73** [comment]
And fixed some minor link text issue in cn site. I think all others are good in CPed changes.

**Comment by nikhilnanivadekar** [comment]
From Google Translate 的源代码 means Source Code. So I had switched the text to be consistent with others. But may be I got it wrong.
https://translate.google.com/?rlz=1C1CHBF_enUS693US693&um=1&ie=UTF-8&hl=en&client=tw-ob#auto/en/%E7%9A%84%E6%BA%90%E4%BB%A3%E7%A0%81

**Comment by itohiro73** [comment]
I think the translation is correct, but pretty much all language sites have a link to "Eclipse Collections" not "source code". So I flipped it back.

**Pull request merged**